async-nats/src/jetstream/object_store/mod.rs

```diff
@@ -850,9 +1066,9 @@ fn is_default<T: Default + Eq>(t: &T) -> bool {
 #[derive(Debug, Default, Clone, Serialize, Deserialize, Eq, PartialEq)]
 pub struct ObjectLink {
     /// Name of the object
-    pub name: String,
+    pub name: Option<String>,
```

Comment:
Presumably this doesn't need to be `Option<String>` if it's already gated behind `AddLinkErrorKind::EmptyName`?

Reply:
It's still a valid case when adding a link to a bucket instead of object.